src/charm.py

```diff
@@ -215,6 +236,34 @@ def _get_relation_id_for_csr(self, csr: str) -> Optional[int]:
             return None
         return relation_ids.pop()
 
+    def _is_certificate_allowed(self, event: CertificateCreationRequestEvent) -> bool:
```

Comment:
Do we need to pass it the whole event?

Reply:
No, I refactored to pass directly the CSR and relation_id instead.